crates/core/src/surfnet/svm.rs

```diff
@@ -937,91 +944,245 @@ impl SurfnetSvm {
     ///
     /// # Arguments
     /// * `slot` - The slot number to retrieve the block for.
+    /// * `config` - The configuration for the block retrieval.
     ///
     /// # Returns
     /// `Some(UiConfirmedBlock)` if found, or `None` if not present.
-    pub fn get_block_at_slot(&self, slot: Slot) -> Option<UiConfirmedBlock> {
-        // Retrieve block
+    pub fn get_block_at_slot(
```

Comment:
This function is quite huge and hard to read, with a good bit of duplicate code. Can we do a quick pass to send some of this logic to smaller functions? `to_ui_inner_instructions`, `to_ui_return_data`, `to_ui_message`, etc etc